src/develop/pixelpipe_cache.c

```diff
@@ -63,7 +63,17 @@ int dt_dev_pixelpipe_cache_init(dt_dev_pixelpipe_cache_t *cache, int entries, si
   return 1;
 
 alloc_memory_fail:
-  dt_dev_pixelpipe_cache_cleanup(cache);
+  //  dt_dev_pixelpipe_cache_cleanup(cache);
+  // The above code seems to be not correct as failing to allocate the cache->data buffers
+  // should not cleanup the whole pixelpipe cache but only reset the buffers to null.
+  // A warning about low memory will appear but the pipeline still has valid data so dt won't crash
+  // but will only fail to generate thumbnails for example.
+  for(int k = 0; k < cache->entries; k++)
+  {
+    dt_free_align(cache->data[k]);
+    cache->size[k] = 0;
+    cache->data[k] = NULL;
+  }
```

Comment:
But aren't we then here leaking memory for:

```
  cache->dsc
  cache->hash
  cache->used
```

Reply:
That's what the old code implied. But that's exactly what i think is wrong.

The allocated memory for `cache->dsc` and friends is deallocated later via
`void dt_dev_pixelpipe_cleanup(dt_dev_pixelpipe_t *pipe)` found in pixelpipe_hb.c.

If we free everything here via `dt_dev_pixelpipe_cache_cleanup` we later have
1. access in the pipeline code to the already deallocated pointers resulting in access violations
2. double freeing in `dt_dev_pixelpipe_cleanup`

The paired cache_init and cache_cleanup has been checked by adding

` fprintf(stderr,"\ndt_dev_pixelpipe_cache_init %p",cache);` and `fprintf(stderr,"\ndt_dev_pixelpipe_cache_cleanup %p",cache);`